### PR TITLE
Add simple fix for swapping record-wrapped owned pointers

### LINKED ISSUE #14120: Unhelpful error message when swapping owned wrappers
### Summary of Problem
Swapping an owned wrapper generates an unhelpful error message.

### Steps to Reproduce

**Source Code:**

```chapel
class Dog { }

record DogOwner {
  var pet: owned Dog?;

  proc init() {
    this.pet = new owned Dog();
  }
}

var p1 = new DogOwner();
var p2 = new DogOwner();

p1 <=> p2;
```

**Compile command:**
```chpl
> chpl dog.chpl

$CHPL_HOME/modules/internal/ChapelBase.chpl:1914: error: const actual is passed to 'ref' formal '_arg2' of =
dog.chpl:3: note: to formal of type DogOwner
dog.chpl:4: note: to ref formal here
```

One can work-around this by defining a swap overload for their record:

```chpl
proc <=>(ref a: DogOwner, ref b: DogOwner) {
  var tmp = a;
  a = b;
  b = tmp;
}
```

### Configuration Information

- Output of `chpl --version`: `1.20.0.c97f29147e`

### PR DESCRIPTION
**Pull request by bradcray**
This adds a variant of a test that BenA introduced in issue #14120
and implements a simple fix for it by changing the temporary used
by the default swap operator from a `const` to a `var`.

Resolves #14120.

### CODE REVIEW

**Comment by bradcray** [comment]
@mppf: Do you have thoughts about the approach taken in this PR to resolve @ben-albrecht's issue, or alternative ways to address it that are more precise / principled?

File: modules/internal/ChapelBase.chpl [hunk 1]
```diff
@@ -1909,7 +1909,12 @@ module ChapelBase {
 
   /* swap operator */
   inline proc <=>(ref lhs, ref rhs) {
-    const tmp = lhs;
+    // it's tempting to make this a `const` tmp, but it causes
+    // problems for cases that self-modify, like records wrapping
```
Comment (mppf) [inline modules/internal/ChapelBase.chpl]: It's not modifying itself - it's modifying `tmp` on the line `rhs = tmp`.

Comment (bradcray) [reply modules/internal/ChapelBase.chpl]: @mppf: How do you feel about the phrasing "but it causes problems for types that self-modify when they appear on the RHS of an assignment, like records wrapping..."?

Comment (mppf) [reply modules/internal/ChapelBase.chpl]: how about

> but it causes problems for types where the RHS of an assignment is modified, as with records containing an `owned` field

The part I was not liking is the phrase "self-modify".

**Pull request merged**